storage/grpc_client.go

```diff
@@ -1762,6 +1774,14 @@ func (w *gRPCWriter) uploadBuffer(recvd int, start int64, doneReading bool) (*st
 				obj = resp.GetResource()
 			}
 
+			// Even though we received the object response, continue reading
+			// until we receive a non-nil error, to ensure the stream does not
+			// leak even if the context isn't cancelled. See:
+			// https://github.com/grpc/grpc-go/commit/365770fcbd7dfb9d921cb44827ede770f33be44f
```

Comment:
This is the actual docs: https://pkg.go.dev/google.golang.org/grpc#ClientConn.NewStream

I'll update it in the comment as well.

Reply:
After looking at these docs I'm wondering if we can use the CloseAndRecv helper instead to avoid writing these loops? We already do this here: https://github.com/googleapis/google-cloud-go/blob/cc83515d0c837c8b1596a97b6f09d519a0f75f72/storage/grpc_client.go#L1748

Reply:
We don't have that helper function for the bidi stream